tools/challenge-editor/api/server.ts

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable @typescript-eslint/no-misused-promises */
+import cors from 'cors';
+import express from 'express';
+import { blockRoute } from './routes/blockRoute';
+import { indexRoute } from './routes/indexRoute';
+import { saveRoute } from './routes/saveRoute';
+import { stepRoute } from './routes/stepRoute';
+import { superblockRoute } from './routes/superblockRoute';
+import { toolsRoute } from './routes/toolsRoute';
+
+(() => {
+  const app = express();
+
+  app.use(
+    cors({
+      origin: 'http://localhost:3300'
+    })
+  );
+
+  app.use(express.static('public'));
+  app.use(express.json());
+  app.use(express.urlencoded());
+
+  app.post('/:superblock/:block/_tools/:command', async (req, res) => {
+    await toolsRoute(req, res);
+  });
+
+  app.post('/:superblock/:block/:step', async (req, res) => {
+    await saveRoute(req, res);
+  });
+
+  app.get('/:superblock/:block/:step', async (req, res) => {
+    await stepRoute(req, res);
+  });
+
+  app.get('/:superblock/:block', async (req, res) => {
+    await blockRoute(req, res);
+  });
+
+  app.get('/:superblock', async (req, res) => {
+    await superblockRoute(req, res);
+  });
+
+  app.get('/', indexRoute);
+
+  app.listen(3200, () => console.log('App is live on 3200!'));
+})();
```

Comment:
```suggestion
const app = express();

app.use(
  cors({
    origin: 'http://localhost:3300'
  })
);

app.use(express.static('public'));
app.use(express.json());
app.use(express.urlencoded());

app.post('/:superblock/:block/_tools/:command', (req, res, next) => {
  toolsRoute(req, res).catch(next);
});

app.post('/:superblock/:block/:step', (req, res, next) => {
  saveRoute(req, res).catch(next);
});

app.get('/:superblock/:block/:step', (req, res, next) => {
  stepRoute(req, res).then(next).catch(next);
});

app.get('/:superblock/:block', (req, res, next) => {
  blockRoute(req, res).catch(next);
});

app.get('/:superblock', (req, res, next) => {
  superblockRoute(req, res).catch(next);
});

app.get('/', indexRoute);

app.listen(3200, () => console.log('App is live on 3200!'));
```
Since this is all synchronous at the top level, there's no need to wrap this in an IIFE.  Also, `async` functions aren't supported by express 4 (apparently they're coming in 5).  It's not a disaster to use them, but any custom error handling won't work.